converter/internal/otelcolconvert/converter_kafkareceiver.go

```diff
@@ -0,0 +1,183 @@
+}
+
+type kafkaReceiverConverter struct{}
+
+func (kafkaReceiverConverter) Factory() component.Factory { return kafkareceiver.NewFactory() }
+
+func (kafkaReceiverConverter) InputComponentName() string { return "" }
+
+func (kafkaReceiverConverter) ConvertAndAppend(state *state, id component.InstanceID, cfg component.Config) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	label := state.FlowComponentLabel()
+
+	args := toKafkaReceiver(state, id, cfg.(*kafkareceiver.Config))
+	block := common.NewBlockWithOverride([]string{"otelcol", "receiver", "kafka"}, label, args)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", stringifyInstanceID(id), stringifyBlock(block)),
+	)
+
+	state.Body().AppendBlock(block)
+	return diags
+}
+
+func toKafkaReceiver(state *state, id component.InstanceID, cfg *kafkareceiver.Config) *kafka.Arguments {
+	var (
+		nextMetrics = state.Next(id, component.DataTypeMetrics)
+		nextLogs    = state.Next(id, component.DataTypeLogs)
+		nextTraces  = state.Next(id, component.DataTypeTraces)
+	)
+
+	return &kafka.Arguments{
+		Brokers:         cfg.Brokers,
+		ProtocolVersion: cfg.ProtocolVersion,
+		Topic:           cfg.Topic,
+		Encoding:        cfg.Encoding,
+		GroupID:         cfg.GroupID,
+		ClientID:        cfg.ClientID,
+		InitialOffset:   cfg.InitialOffset,
+
+		Authentication:   toKafkaAuthentication(encodeMapstruct(cfg.Authentication)),
+		Metadata:         toKafkaMetadata(cfg.Metadata),
+		AutoCommit:       toKafkaAutoCommit(cfg.AutoCommit),
+		MessageMarking:   toKafkaMessageMarking(cfg.MessageMarking),
+		HeaderExtraction: toKafkaHeaderExtraction(cfg.HeaderExtraction),
+
+		Output: &otelcol.ConsumerArguments{
+			Metrics: toTokenizedConsumers(nextMetrics),
+			Logs:    toTokenizedConsumers(nextLogs),
+			Traces:  toTokenizedConsumers(nextTraces),
+		},
+	}
+}
+
+func toKafkaAuthentication(cfg map[string]any) kafka.AuthenticationArguments {
+	spew.Dump(cfg)
+
+	return kafka.AuthenticationArguments{
+		Plaintext: toKafkaPlaintext(encodeMapstruct(cfg["plain_text"])),
+		SASL:      toKafkaSASL(encodeMapstruct(cfg["sasl"])),
+		TLS:       toKafkaTLSClientArguments(encodeMapstruct(cfg["tls"])),
+		Kerberos:  toKafkaKerberos(encodeMapstruct(cfg["kerberos"])),
+	}
+}
+
+func toKafkaPlaintext(cfg map[string]any) *kafka.PlaintextArguments {
+	if cfg == nil {
+		return nil
+	}
+
+	return &kafka.PlaintextArguments{
+		Username: cfg["username"].(string),
+		Password: rivertypes.Secret(cfg["password"].(string)),
+	}
+}
+
+func toKafkaSASL(cfg map[string]any) *kafka.SASLArguments {
+	if cfg == nil {
+		return nil
+	}
+
+	return &kafka.SASLArguments{
+		Username:  cfg["username"].(string),
+		Password:  rivertypes.Secret(cfg["password"].(string)),
+		Mechanism: cfg["mechanism"].(string),
+		Version:   cfg["version"].(int),
+		AWSMSK:    toKafkaAWSMSK(encodeMapstruct(cfg["aws_msk"])),
+	}
+}
+
+func toKafkaAWSMSK(cfg map[string]any) kafka.AWSMSKArguments {
+	return kafka.AWSMSKArguments{
```

Comment:
are we missing a nil check at the start of this one or is it unneeded here?

Reply:
I'll add one to be safe but it shouldn't be necessary since it's a non-pointer in OpenTelemetry; it'll always be available.